### ISSUE TITLE
Bulk exclude backlogged reportbacks on node 362

### ISSUE THREAD

**Issue by mikefantini** [DoSomethingArchive/legacy-website#6344]
#### User Story

As an admin, I want all fids on node 362 with a quantity under 200 to be marked as excluded before it relaunches on Thursday.
#### Additional Information (optional)

Campaigns team was trying to go through the backlog and the number of reportbacks went from ~600 to ~1500. Got these numbers from Lizzy. 600 felt doable, 1500 seems unreasonable between now and then. Also a few reports of "Illegal choice detected" error when intern was running through things.
#### Why This Matters

Since campaign relaunches on Thursday and reportbacks are listed as oldest to newest the campaigns team won't realistically be able to get to current reportbacks without this being cleared out. Only other option I can think of is that we adjust the script to include only reportbacks that match one of the previous run ids which would buy us a little time I suppose and allow us to get the work done later this week.

@angaither let's chat about this in the AM

**Comment by chloealee** [comment]
@mikefantini I'm not sure what you mean by all "fids with a quantity under 200 to be marked as excluded" - isn't quantity attached to reportbacks, not reportback items? 

Also, admins review these reportbacks at `/node/362/rb`?

**Comment by mikefantini** [comment]
@chloealee right sorry I guess the FIDs are what need to be marked as excluded and we should do that for any that are currently pending and associated with RBIDs with a quantity under 200

**Comment by namimody** [comment]
From what I understand, four things need to happen:
- all FIDs associated with a RB with a quantity under 200 get marked as "excluded"
- these photos no longer appear in the queue
- these RBIDs should have a status of 0 in the DB
- FIDs associated with RBIDs with a quantity 201 and over should still be pending and appear in the inbox with a status of NULL

Might also be worth checking if there are any RBIDs with some photos that have been reviewed and some still in the inbox so that this fix works for those edge cases.

![image](https://cloud.githubusercontent.com/assets/12396076/14290007/5e50151a-fb2b-11e5-9100-2381b22500e7.png)

**Comment by mikefantini** [comment]
@namimody @chloealee one caveat to the above, we only want FIDs that are currently pending and quantity under 200 to be marked as excluded.

i.e. if an FID is already approved, we don't want it to change to be excluded.

**Comment by mikefantini** [comment]
Also if the RBID is already 1 we don't it to change to 0.

**Comment by namimody** [comment]
Thanks @mikefantini! So then this work should only touch RBIDs with status NULL? cc @chloealee

**Comment by chloealee** [comment]
@mikefantini @namimody should this be ONLY for node 362 or for all campaign nodes?

**Comment by namimody** [comment]
Only 362 @chloealee!

**Comment by chloealee** [comment]
cool, thanks @namimody !

**Comment by mikefantini** [comment]
@namimody I'm pretty sure that flagged status can stay 0 or 1 for a file that is pending. Consider if it's a second file on an existing reportback that had a file that was previously reviewed.

i.e. fid 1 is approved and rbid flagged status is now 0. fid 2 (2nd file on the same reportback) is pending but rbid flagged status is still 0. So I'm pretty sure you need to look for all files with status pending regardless of whether flagged status is 0, 1, or NULL @chloealee

**Comment by namimody** [comment]
Ohh now I follow @mikefantini -- so even if we make mark files of partially reviewed reportbacks as excluded, their RBID status will stay 0 or 1. Therefore, I think should be comprehensive?
- all not yet reviewed FIDs associated with a RB with a quantity under 200 get marked as "excluded"
- these photos should no longer appear in the queue
- these RBIDs should have a status of 0 or 1 in the DB (depending on if other assoicated FIDs have been reviewed and been marked as approved/excluded/promoted or flagged previously)
- unreviewed FIDs associated with RBIDs with a quantity 201 and over should still be pending and appear in the inbox

cc @chloealee

**Comment by angaither** [comment]
this failed on thor

```
Dosomething_reportback  7034  Bulk excludes backlogged reportbacks on node   
                               362.  Updates any fid with a pending status to 
                               be marked as excluded if the quantity in the   
                               reportback is less than 200.
Do you wish to run all pending updates? (y/n): 
y
array_flip(): Can only flip STRING and INTEGER values!               [warning]
entity.controller.inc:219
PHP Fatal error:  Call to a member function getFids() on a non-object in /var/www/www.dosomething.org/releases/20160406175520/lib/modules/dosomething/dosomething_reportback/dosomething_reportback.install on line 935
Fatal error: Call to a member function getFids() on a non-object in /var/www/www.dosomething.org/releases/20160406175520/lib/modules/dosomething/dosomething_reportback/dosomething_reportback.install on line 935
```

**Comment by chloealee** [comment]
@angaither I'm getting these errors now too:

```
WD Apache Solr: HTTP Status: 0; Message: : php_network_getaddresses: getaddrinfo failed: Name or service not known; Response: ; Request: Unknown; Caller: apachesolr_views_views_data() (line 43 [error]
of /var/www/dev.dosomething.org/html/profiles/dosomething/modules/contrib/apachesolr_views/apachesolr_views.views.inc)
WD Apache Solr Views: HTTP 0; : php_network_getaddresses: getaddrinfo failed: Name or service not known
```

Do you know why these files might be triggered?

**Comment by angaither** [comment]
@chloealee where are you seeing this?

**Comment by chloealee** [comment]
@angaither when I run `drush up db -y` this is the only error now returning (I got rid of the other error because I realized when I was cleaning up the code, I accidentally deleted `rb.rbid` from the `SELECT` statement).

**Comment by chloealee** [comment]
@angaither I got everything to work but that error still pops up - is it okay to submit another PR even with this error?

**Comment by angaither** [comment]
@chloealee yeah

**Comment by mikefantini** [comment]
Just leaving some notes here:

First the great news: my new reportbacks not on the run were not touched. Woo.

Now some funkiness:

When I went to review reportbacks on node 362 it initially seemed like there were still thousands of images:

![image](https://cloud.githubusercontent.com/assets/6330971/14335363/8d8ae104-fc28-11e5-8641-d18342e28979.png)

I then went to actually review what was left in the inbox. 

Reviewed 25 in the inbox. Got a 503. Refreshed the page. Inbox now 91. 

Reviewed 25 in the inbox. Got a 503. Refreshed the page. Inbox now 66. So all 25 reviewed reportbacks went through. There were several illegal choice detected errors.

Reviewed 25 in the inbox. Got a 503. Refreshed the page. Inbox now 44. So 22 reviewed reportbacks went through and 3 reportbacks that I had already reviewed were still in the inbox with a radio button selected. There were 21 illegal choice detected errors.

Reviewed 25 in the inbox. Got a 503. Refreshed the page. Inbox now 41. So 3 reviewed reportbacks went through and 22 photos I had already reviewed was still in the inbox with a radio button selected. And there were 3 illegal choice detected errors 

Reviewed 25 in the inbox. Got a 503. Refreshed the page. Inbox now 17. So 24 reviewed reportbacks went through and 1 photo I had previously reviewed was still in inbox with radio button selected. 16 illegal choice detected.
